src/context.ts

```diff
@@ -1,5 +1,14 @@
 import {createContext} from 'react';
+import {MyAppContext} from '../';
 
-const Context = createContext(({} as any));
+const defaultAuth = {
+  user: undefined,
+  access: false,
```

Comment:
Just wanted to point out that these changes affect `_app.tsx` in the way the user experiences the site. While the backend does its work, the state of access and redirect are not `true` or `false` but null - so the loader is rendered. Once the state is updated to `true` or `false` then we render or do the appropriate function based on the page itself. 

![image](https://user-images.githubusercontent.com/3924990/72552351-88614480-384b-11ea-82f3-dad2bd39b910.png)